questionnaire/submissions-service.js

```diff
@@ -0,0 +1,42 @@
+'use strict';
+
+const createMessageBusCaller = require('../services/message-bus');
+const createQuestionnaireDAL = require('./questionnaire-dal');
+
+function createSubmissionService({logger}) {
+    const db = createQuestionnaireDAL({logger});
+
+    async function getQuestionnaireIdsBySubmissionStatus(status) {
+        return db.getQuestionnaireIdsBySubmissionStatus(status);
+    }
+
+    async function updateQuestionnaireSubmissionStatus(questionnaireId, submissionStatus) {
+        return db.updateQuestionnaireSubmissionStatus(questionnaireId, submissionStatus);
+    }
+
+    async function postFailedSubmissions() {
+        const questionnaireIds = await getQuestionnaireIdsBySubmissionStatus('FAILED');
+        let response = [];
+        const messageBusCaller = createMessageBusCaller({logger});
+        const promises = questionnaireIds.map(async id => {
+            await updateQuestionnaireSubmissionStatus(id, 'IN_PROGRESS');
+            let promise;
+            try {
+                promise = messageBusCaller.post('SubmissionQueue', {
+                    applicationId: id
+                });
+            } catch (err) {
+                await updateQuestionnaireSubmissionStatus(id, 'FAILED');
```

Comment:
This won't be hit

Reply:
coverage says it is

Reply:
I assume that's because the test is wrong? https://github.com/CriminalInjuriesCompensationAuthority/data-capture-service/pull/199#discussion_r644903934